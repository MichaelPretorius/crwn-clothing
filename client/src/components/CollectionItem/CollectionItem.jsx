import React from 'react';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cartActions';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  BackgroundImage,
  NameContainer,
  PriceContainer,
  ButtonContainer
} from './CollectionItem.styles';

const CollectionItem = ({ item, addItem }) => {
  const { name, imageUrl, price } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>R{price}</PriceContainer>
      </CollectionFooterContainer>
      <ButtonContainer inverted onClick={() => addItem(item)}>
        Add To Cart
            </ButtonContainer>
    </CollectionItemContainer>
  );
}

export default connect(null, { addItem })(CollectionItem);