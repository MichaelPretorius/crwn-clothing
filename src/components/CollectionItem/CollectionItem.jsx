import React from 'react';
import { connect } from 'react-redux';

import './CollectionItem.scss';
import CustomButton from '../CustomButton/CustomButton';
import { addItem } from '../../redux/cart/cartActions';

const CollectionItem = ({ item, addItem }) => {
    const { name, imageUrl, price } = item;

    return (
        <div className="collection-item">
            <div className="image" style={{
                backgroundImage: `url(${imageUrl})`
            }} />
            <div className="collection-footer">
                <span className="name">{name}</span>
                <span className="price">R{price}</span>
            </div>
            <CustomButton inverted onClick={() => addItem(item)}>
                Add To Cart
            </CustomButton>
        </div>
    );
}

export default connect(null, { addItem })(CollectionItem);