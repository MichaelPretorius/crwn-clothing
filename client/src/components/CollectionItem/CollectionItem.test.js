import React from 'react';
import { shallow } from 'enzyme';

import { CollectionItem } from './CollectionItem';

describe('CollectionItem', () => {
  let wrapper;
  let mockAddItem;
  const imageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: imageUrl,
        price: mockPrice,
        name: mockName
      },
      addItem: mockAddItem
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  test('should render CollectionItem component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
  
  test('should call addItem when AddButton clicked', () => {
    wrapper.find('ButtonContainer').simulate('click');
    expect(mockAddItem).toBeCalled();
  });
  
  test('should render imageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toEqual(imageUrl)
  })
  
  test('should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toEqual(mockName);
  });
  
  test('should render price prop in PriceContainer', () => {
    const price = wrapper.find('PriceContainer').text();
    expect(price).toEqual(`R${mockPrice}`);
  });
})
