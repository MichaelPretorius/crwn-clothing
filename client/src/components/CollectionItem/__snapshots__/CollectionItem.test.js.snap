// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem should render CollectionItem component 1`] = `
"<CollectionItemContainer>
  <BackgroundImage className=\\"image\\" imageUrl=\\"www.testImage.com\\" />
  <CollectionFooterContainer>
    <NameContainer>
      black hat
    </NameContainer>
    <PriceContainer>
      R
      10
    </PriceContainer>
  </CollectionFooterContainer>
  <ButtonContainer inverted={true} onClick={[Function: onClick]}>
    Add To Cart
  </ButtonContainer>
</CollectionItemContainer>"
`;
